hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,171 @@
+Each of the four interfaces can request data such that any `genbits` bus return can supply any requesting peripheral block.
+
+A configurable feature for peripheral port 0 is that it can be put into broadcast mode.
+This means that the EDN block will request for that port and is paced by a configurable timer.
+Data will be delivered continuously on the bus for any snooping participant to consume.
+
+Commands to the CSRNG block can be generated by either firmware or hardware.
+For firmware generated commands, a TLUL set of registers will allow any CSRNG application interface command to be sent to CSRNG.
+For hardware generated commands, there are two modes that allow this.
+One is the auto request mode, where two FIFOs are provided to send commands and additional data. 
+The general operation of this mode is that the CSRNG instance is set up by firmware, then the FIFOS are preloaded with commands.
+One FIFO can be programmed to send `generate` commands.
+The other FIFO can be programmed to send `reseed` commands.
+There is a configurable register that sets the number of requests between reseeds.
+This register determines how many `generate` commands are sent before a `reseed` command is sent.
+The process of sending these two commands will repeat forever until the module is reset or the enable bit is cleared.
+
+## Example Topology
+The diagram below shows an example topology where two EDN modules are used to distribute CSRNG `genbits` to peripheral modules.
+
+![EDN Example Topology Diagram](edn_top_diag.svg)
+
+
+## Compatibility
+Commands generated toward the CSRNG application interface must conform to the CSRNG application interface specification.
+
+
+# Theory of Operations
+
+The first step that must be done to enable the EDN block is to configure the  {{< regref "CTRL" >}} register.
+This will enable the block in general, and then determine which of three modes are to be enabled.
+Broadcast mode can be selected independent of auto request mode or boot request mode.
+A choice between auto request mode and boot request mode should be made.
+
+Once a CSRNG command request mode has been configured, the EDN block will start requesting to the CSRNG block.
+Data will be returned from the specific CSRNG instance that the EDN block is connected to.
+It is assumed that the EDN block has sent an `instantiate` application command before sending any `generate` application commands.
+
+Any peripheral module that is on a request/ack bus should expect the EDN block to fetch `genbits` only when requested.
+Conversely, any peripheral module that is on a broadcast bus should expect the EDN block to fetch `genbits` continuously without a request.
+
+When in auto request mode, the {{< regref "SUM_STS" >}} register will indicate when this mode is done, as in the state machine is idle.
+For very long requests, this indication will be always on unless the module is disabled.
+
+## Block Diagram
+
+![EDN Block Diagram](edn_blk_diag.svg)
```

Comment:
The signals coming out of EDN registers overlap with the boxes.

There are also some phantom grey lines floating around: I think they need deleting.

Finally, the bounding box needs sorting out: there's a big white gap underneath the image.

Reply:
I do not see these issues on my version of rendering, except for the white gap of whch i thought was markdown issue.

Reply:
I just opened up the SVG in Inkscape. The "grey lines" are actually light blue (RGB `#e8ebf4`). For example, there's one with ID `path79`. I don't know whether this is a problem with Visio export?

I assume you're doing this already, but just in case... You can generate the documentation locally with `./util/build_docs.py --preview` and then browse to `http://localhost:1313`. If you do so, you can look at just the block diagram by going to `http://localhost:1313/hw/ip/edn/doc/edn_blk_diag.svg`. (Or you could just load it based on the path on disk: Hugo doesn't do anything magic to SVGs).

Reply:
thank @rswarbrick for checking this. I am doing the process you described on my WSL ubuntu window. Our previous process was unsatisfactory. I looked again and just do not see these colors lines. Again maybe @martin-lueker can re-create this problem with his system.

Reply:
Interesting.  Hi @rswarbrick, I'm going through all of @mwbranstad's open PR's now.  I'll take a look at the original visio.

Reply:
Hi @rswarbrick, I see your second two points, and think I can definitely fix those.   However, your observation about the signal/box overlap is simply not jumping out at me either, neither in Inkscape, nor in the preview.   Is there any way you could send me/post a screenshot highlighting this?

Reply:
BTW, since you have given such excellent attention to this figure, do you have any rendering comments on the top-level figure?

Reply:
Hi @rswarbrick This is indeed a great review.  I think I've fixed the image.  Could you please take a look?  (BTW, I think once we have your sign-off this may be ready to go in).

Reply:
The block diagram looks much better, thanks. Unfortunately, the text still overlaps. I think the problem is probably that the font is specified as Calibri, which isn't installed on my Linux box. Looking at the result, I think you can make things work even when the viewer doesn't have a font with compatible metrics by:

- Right-align the text in the signals coming out of the EDN registers block, so the ragged edge grows to the left
- Slightly widen the boxes for the two FIFOs and "EDN Main SM"
- Right-align the "End Point Interface <n>" text
- (optional) Centre the text you've got above boxes, such as "flops" and "128 to N unpacker". This doesn't crash into anything, but it looks a bit weird if it falls off one edge.

I also don't think that the end point interfaces are spaced equally, or maybe there's a problem with the text spacing above the unpackers? It looks like the text above the unpacker for interface 2 is much closer to the box above than the others.

To check this locally, I'd suggest exporting to SVG and then opening up in a text editor and replacing "calibri" with (e.g.) "Cabbage". Assuming you don't have a cabbage font installed, you'll probably get something like what I'm seeing. (Probably don't upload the brassica'd version :-))

At the moment, the diagram looks like this to me:

![ss](https://user-images.githubusercontent.com/104845/93315951-88af4680-f803-11ea-8095-040f40c526c4.png)

Reply:
Ha, and here I though I was safe rendering it on a Linux machine, and viewing it from a browser on windows.  Good grief!  Will look at this again, maybe pick a common font.